test/integration/testpaths/testpath_multiple_snapshot.py

```diff
@@ -0,0 +1,255 @@
+
+        cls.skiptest = False
+
+        clus_list = list_clusters(cls.apiclient)
+
+        if cls.hypervisor.lower() not in ['vmware'] or len(clus_list) < 2:
+            cls.skiptest = True
+            return
+
+        try:
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+            # Create Service offering
+            cls.service_offering_zwps = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+                tags=ZONETAG1
+            )
+
+            cls.disk_offering_zwps = DiskOffering.create(
+                cls.apiclient,
+                cls.testdata["disk_offering"],
+                tags=ZONETAG1
+            )
+
+            cls._cleanup = [
+                cls.account,
+                cls.service_offering_zwps,
+                cls.disk_offering_zwps,
+            ]
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+
+        self.cleanup = []
+        if self.skiptest:
+            self.skipTest("Skip test as setup is either not VMWare or  \
+                    having less than 2 clusters %s" % self.hypervisor)
+
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.pools = []
+
+    def tearDown(self):
+        try:
+            for storagePool in self.pools:
+                StoragePool.update(self.apiclient, id=storagePool.id, tags="")
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="true")
+    def test_01_multiple_snapshot_in_zwps(self):
+        """ Test multiple volume snapshot in zwps
+
+        # 1. Verify if setup has a ZWPS and 2 CWPS
+        # 2. Deploy a VM with data disk in ZWPS
+        # 1. Verify ROOT and DATA Disk of the VM is in ZWPS.
+        # 2. Take a snapshot of VM.
+        # 3. Create Multiple Snapshots till operation fails.
+        """
+        try:
+            if (self.clus_list) < 2:
+                self.skipTest(
+                    "There must be at least two clusters in the setup")
+
+            self.pools = StoragePool.list(self.apiclient, zoneid=self.zone.id)
```

Comment:
We are skipping the test here if the self.clus_list < 2. In that case self.pools will not have any value. However, we are referring it in teatDown method. I think test will fail in that case.
We are already validating the no.of clusters in setUpClass method. So why do we need this here?